src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -798,6 +803,102 @@ public static String getSinceVersion(String moduleName, DetailNode moduleJavadoc
         return sinceVersion;
     }
 
+    /**
+     * Gets the specifically indicated version of module's property from the javadoc of module.
+     *
+     * @param propertyName the name of property.
+     * @param moduleJavadoc the javadoc of module.
+     * @return the specific since version of module's property.
+     */
+    @Nullable
+    private static String getSpecifiedPropertyVersion(String propertyName,
+                                                      DetailNode moduleJavadoc) {
+        String specifiedVersion = null;
+
+        final DetailNode propertyModuleJavadoc =
+            getPropertyJavadocNodeInModule(propertyName, moduleJavadoc);
+
+        final DetailNode primaryJavadocInlineTag = JavadocUtil.findFirstToken(propertyModuleJavadoc,
+                        JavadocTokenTypes.JAVADOC_INLINE_TAG);
+
+        for (DetailNode textNode = JavadocUtil
+            .getNextSibling(primaryJavadocInlineTag, JavadocTokenTypes.TEXT);
+             textNode != null && specifiedVersion == null;
+             textNode = JavadocUtil.getNextSibling(
+                 textNode, JavadocTokenTypes.TEXT)) {
```

Comment:
Do you mean .... That you need list of text nodes ?
Better to have method `  getAllNodesOfType(primaryJavadocInlineTag, TEXT)`
And for just loop over collection, with filtering .

Reply:
all optimizations for implementation will be done in https://github.com/checkstyle/checkstyle/issues/17256 , we  need merge this PR sooner, to unblock all other activitites for version fixes.